specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -16,119 +17,119 @@ enum TextCategory {
+  @maxLength(10000)
   text: string;
 
-  @doc("The categories will be analyzed. If not assigned, a default set of the categories' analysis results will be returned.")
+  @doc("The categories will be analyzed. If they are not assigned, a default set of analysis results for the categories will be returned.")
   categories?: TextCategory[];
 
   @doc("The names of blocklists.")
   blocklistNames?: string[];
 
   @doc("When set to true, further analyses of harmful content will not be performed in cases where blocklists are hit. When set to false, all analyses of harmful content will be performed, whether or not blocklists are hit.")
-  breakByBlocklists?: boolean;
+  haltOnBlocklistHit?: boolean;
+
+  @doc("This refers to the type of text analysis output. If no value is assigned, the default value will be \"FourSeverityLevels\".")
+  outputType?: AnalyzeTextOutputType = AnalyzeTextOutputType.FourSeverityLevels;
 }
 
-@doc("The analysis response of the text")
+@doc("The text analysis response.")
 model AnalyzeTextResult {
-  @doc("The details of blocklist match.")
-  blocklistsMatchResults?: TextBlocklistMatchResult[];
-
-  @doc("Analysis result for Hate category.")
-  hateResult?: TextAnalyzeSeverityResult;
-
-  @doc("Analysis result for SelfHarm category.")
-  selfHarmResult?: TextAnalyzeSeverityResult;
+  @doc("The blocklist match details.")
+  blocklistsMatch?: TextBlocklistMatch[];
 
-  @doc("Analysis result for Sexual category.")
-  sexualResult?: TextAnalyzeSeverityResult;
-
-  @doc("Analysis result for Violence category.")
-  violenceResult?: TextAnalyzeSeverityResult;
+  @doc("Analysis result for categories.")
+  categoriesAnalysis: TextCategoriesAnalysis[];
 }
 
 @doc("The result of blocklist match.")
-model TextBlocklistMatchResult {
-  @doc("The name of matched blocklist.")
+model TextBlocklistMatch {
+  @doc("The name of the matched blocklist.")
   @maxLength(64)
   blocklistName: string;
 
-  @doc("The id of matched item.")
+  @doc("The ID of the matched item.")
   @maxLength(64)
-  blockItemId: string;
+  blocklistItemId: string;
 
-  @doc("The content of matched item.")
+  @doc("The content of the matched item.")
   @maxLength(128)
-  blockItemText: string;
-
-  @doc("The character offset of matched text in original input.")
-  offset: int32;
-
-  @doc("The length of matched text in original input.")
-  length: int32;
+  blocklistItemText: string;
 }
 
 @doc("Text analysis result.")
-model TextAnalyzeSeverityResult {
-  @doc("The text category.")
+model TextCategoriesAnalysis {
+  @doc("The text analysis category.")
   category: TextCategory;
 
-  @doc("The higher the severity of input content, the larger this value is. The values could be: 0,2,4,6.")
-  severity: int32;
+  @doc("The value of this field is determined by the outputType specified in the request. If 'FourSeverityLevels' or 'EightSeverityLevels' is chosen, this field will be included in the output. The value increases with the severity of the input content.")
```

Comment:
I think it would better to describe the meaning of this field first, then describe how it is affected by outputType..
```suggestion
  @doc("The value increases with the severity of the input content. The value of this field is determined by the outputType specified in the request. If 'FourSeverityLevels' or 'EightSeverityLevels' is chosen, this field will be included in the output.")
```

Reply:
Make sense! Fixed according to your suggestion.